File: src/CitiesOnMap.Domain/Entities/City.cs
namespace CitiesOnMap.Domain.Entities;

public class City
{
    public string Id { get; set; } = null!;
    public string Name { get; set; } = string.Empty;
    public string NameAscii { get; set; } = string.Empty;
    public decimal Latitude { get; set; }
    public decimal Longitude { get; set; }
    public int CountryId { get; set; }
    public Country Country { get; set; } = null!;
    public string AdministrationName { get; set; } = string.Empty;
    public CapitalType CapitalType { get; set; }
    public int Population { get; set; }
}

File: src/CitiesOnMap.WebUI/src/app/_models/game/city.ts
import {Country} from "./country";
import {CapitalType} from "../../_enums/capitalType";

export interface City {
  id: string;
  name: string;
  nameAscii: string;
  latitude: number;
  longitude: number;
  countryId: number;
  country: Country;
  administrationName: string;
  capitalType: CapitalType;
  population: number;
}
